[builder error, cgen] access to a pointer from a generic function

### Describe the bug

```v
module main

struct Foo123 {
	field string
}

fn gen_func[T](value T) string {
	$if T is i32 {
		return "123"
	}
	$else $if T is &Foo123 {
		return value.field
	}
	$else {
		return "123"
	}
}

fn main() {
	gen_func[&Foo123](&Foo123{})
}
```
```c
VV_LOCAL_SYMBOL string main__gen_func_T___ptr__main__Foo123(main__Foo123* value) {
	#if 121 == 7 && false == false
	{
	}
	#elif 121 == 121 && false == false
	{
		return value.field;
	}
	#endif
	return (string){.str=(byteptr)"", .is_lit=1};
}
```

### Reproduction Steps

`v run bug5.v`

### Expected Behavior

ok

### Current Behavior

```
================== C compilation error (from tcc): ==============
cc: C:/Users/russo/AppData/Local/Temp/v_0/bug5.01JDJCEFA8CHGD3TJT5HXN60MT.tmp.c:7172: warning: implicit declaration of function 'tcc_backtrace'
cc: C:/Users/russo/AppData/Local/Temp/v_0/bug5.01JDJCEFA8CHGD3TJT5HXN60MT.tmp.c:13878: error: expected struct or union but not 'struct main__Foo123 *'
... (the original output was 3 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.8 95e5ba4

### Environment details (OS name and version, etc.)

V full version: V 0.4.8 a132d97.95e5ba4
OS: windows,  Windows 10 Pro v19042 64-bit
Processor: 16 cpus, 64bit, little endian, 

getwd: C:\Users\russo\Desktop\vbugs
vexe: M:\_dev\vlang\v.exe
vexe mtime: 2024-11-25 19:17:14

vroot: OK, value: M:\_dev\vlang
VMODULES: OK, value: C:\Users\russo\.vmodules
VTMP: OK, value: C:\Users\russo\AppData\Local\Temp\v_0

Git version: git version 2.46.0.windows.1
Git vroot status: weekly.2024.47-38-g95e5ba44
.git/config present: true

CC version: Error: 'cc' is not recognized as an internal or external command,
operable program or batch file.

emcc version: Error: 'emcc' is not recognized as an internal or external command,
operable program or batch file.

thirdparty/tcc: N/A
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -4278,7 +4278,7 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 	}
 	alias_to_ptr := sym.info is ast.Alias && sym.info.parent_type.is_ptr()
 	if field_is_opt
-		|| ((node.expr_type.is_ptr() || sym.kind == .chan || alias_to_ptr)
+		|| ((g.unwrap_generic(node.expr_type).is_ptr() || sym.kind == .chan || alias_to_ptr)
 		&& node.from_embed_types.len == 0)
 		|| (node.expr.is_as_cast() && g.inside_smartcast) {
 		g.write('->')

--- a/vlib/v/tests/generic_selector_ptr_test.v
+++ b/vlib/v/tests/generic_selector_ptr_test.v
@@ -1,0 +1,19 @@
+module main
+
+struct Foo123 {
+	field string = 'foobar'
+}
+
+fn gen_func[T](value T) string {
+	$if T is i32 {
+		return '123'
+	} $else $if T is &Foo123 {
+		return value.field
+	} $else {
+		return '123'
+	}
+}
+
+fn test_main() {
+	assert gen_func[&Foo123](&Foo123{}) == 'foobar'
+}

